packages/perseus/src/interactive2/__tests__/movable.test.ts

```diff
@@ -0,0 +1,182 @@
+/* eslint-disable testing-library/prefer-user-event */
+import {fireEvent} from "@testing-library/react";
+
+import {Graphie} from "../../util/graphie";
+import Movable from "../movable";
+
+// Yay for side-effect imports!
+// eslint-disable-next-line import/no-unassigned-import
+import "../../util/interactive";
+
+describe("Movable", () => {
+    it("should be constructable", () => {
+        const movable = new Movable(
+            new Graphie(document.createElement("div")),
+            {},
+        );
+
+        expect(movable).toBeDefined();
+    });
+
+    it("should have accessor properties", () => {
+        const movable = new Movable(
+            new Graphie(document.createElement("div")),
+            {},
+        );
+
+        expect(typeof movable.isHovering).toBe("function");
+        expect(typeof movable.cursor).toBe("function");
+        expect(typeof movable.added).toBe("function");
+        expect(typeof movable.isHovering).toBe("function");
+        expect(typeof movable.isMouseOver).toBe("function");
+        expect(typeof movable.isDragging).toBe("function");
+        expect(typeof movable.mouseTarget).toBe("function");
+    });
+
+    it("cloneState()", () => {
+        const movable = new Movable(
+            new Graphie(document.createElement("div")),
+            {},
+        );
+
+        expect(movable.cloneState()).toStrictEqual(
+            expect.objectContaining({
+                id: expect.any(String),
+                add: [],
+                added: true,
+                cursor: null,
+                draw: [],
+                isDragging: false,
+                isHovering: false,
+                isMouseOver: false,
+                modify: [],
+                mouseTarget: null,
+                onClick: [],
+                onMove: [],
+                onMoveEnd: [],
+                onMoveStart: [],
+                remove: [],
+            }),
+        );
+    });
+
+    it("cloneState() should copy state", () => {
+        const movable = new Movable(
+            new Graphie(document.createElement("div")),
+            {},
+        );
+
+        expect(movable.cloneState()).not.toBe(movable.state);
+    });
+
+    it("modify() resets movable to initial state", () => {
+        const movable = new Movable(
+            new Graphie(document.createElement("div")),
+            {},
+        );
+
+        const state = movable.cloneState();
+        movable.state.added = false;
+
+        movable.modify();
+
+        expect(movable.cloneState()).toStrictEqual(state);
```

Comment:
Is `toStrictEqual` needed here? Would `toEqual` be correct?

Reply:
:TIL: about the differences.  `toStrictEqual` verifies types as well, but I think `.toEqual()` is fine.